diff -Naur a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2015-06-16 17:35:08.000000000 +0200
+++ b/src/Makefile.am	2015-07-16 18:15:31.171794100 +0200
@@ -51,7 +51,7 @@
         rpc_callmsg.c rpc_generic.c rpc_soc.c rpcb_clnt.c rpcb_prot.c \
         rpcb_st_xdr.c svc.c svc_auth.c svc_dg.c svc_auth_unix.c svc_auth_none.c \
         svc_generic.c svc_raw.c svc_run.c svc_simple.c svc_vc.c getpeereid.c \
-        auth_time.c auth_des.c authdes_prot.c debug.c
+        auth_time.c debug.c
 
 ## XDR
 libtirpc_la_SOURCES += xdr.c xdr_rec.c xdr_array.c xdr_float.c xdr_mem.c xdr_reference.c xdr_stdio.c
diff -Naur a/src/rpc_soc.c b/src/rpc_soc.c
--- a/src/rpc_soc.c	2015-06-16 17:35:08.000000000 +0200
+++ b/src/rpc_soc.c	2015-07-16 18:14:51.157595300 +0200
@@ -524,6 +524,7 @@
  * Create the client des authentication object. Obsoleted by
  * authdes_seccreate().
  */
+#ifdef NEVER
 AUTH *
 authdes_create(servername, window, syncaddr, ckey)
 	char *servername;		/* network name of server */
@@ -551,6 +552,7 @@
 	dummy = authdes_seccreate(servername, window, NULL, ckey);
 	return (dummy);
 }
+#endif
 
 /*
  * Create a client handle for a unix connection. Obsoleted by clnt_vc_create()
