diff -Naur a/config.mk b/config.mk
--- a/config.mk	2015-02-19 17:51:23.000000000 +0100
+++ b/config.mk	2015-07-07 17:54:45.626756100 +0200
@@ -10,7 +10,7 @@
 TERMINFO := ${DESTDIR}${PREFIX}/share/terminfo
 
 INCS = -I.
-LIBS = -lc -lutil -lncursesw
+LIBS = -lc -lutil -lncurses
 CPPFLAGS = -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED
 CFLAGS += -std=c99 ${INCS} -DVERSION=\"${VERSION}\" -DNDEBUG ${CPPFLAGS}
 LDFLAGS += ${LIBS}
